### PR TITLE
[4.0] Namespace - Change default controller

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue #15041.

### Summary of Changes
This PR changes default controller name from COM_NAME to a generic **controller** name as discussed in #15041. I also make some small clean up to the code which can be easily seen while review

### Testing Instructions
Code review. Ping @wilsonge and @mbabker for final decision

### CODE REVIEW

**Comment by joomdonation** [comment]
While looking at this, I also have some questions:

1. With the current implement, the data stored in namespace column must contain backslash Joomla\Component\Content\ . I think without backlash Joomla\Component\Content might be better

2. Do we really want to introduce a new way to pass controller name via controller variable in the request? At least for current usage, the widely use task=controller.method is good enough.

File: libraries/src/Cms/Controller/Admin.php [hunk 1]
```diff
@@ -3,8 +3,8 @@
  * @package     Joomla.Cms
  * @subpackage  Controller
  *
- * @copyright  Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
- * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ * @copyright   Copyright (C) 2005 - 2017 Open Source Matters, Inc. All rights reserved.
```
Comment (laoneo) [inline libraries/src/Cms/Controller/Admin.php]: Why this changes, only 2 spaces should be there, otherwise you get CS errors.

Comment (joomdonation) [inline libraries/src/Cms/Controller/Admin.php]: It was CS error before I made the change :(

Comment (laoneo) [inline libraries/src/Cms/Controller/Admin.php]: It must be only two spaces, when you start to revert all of that, then we need to redo this changes again. I don't know why this gives errors on j4, but on 3.8 it is correct and that's how the flow should be staging -> 3.8 -> 4.0.

Comment (joomdonation) [inline libraries/src/Cms/Controller/Admin.php]: Someone who setup code style rules for 4.0-dev branch has to check it then. Otherwise, every PR to 4.0-dev will get this codestyle error.

However, quick look at it again, maybe 3 space is expected change:

1. It make the copyright text align property with the text in @package and @subpackage tag

2. Look at this file https://github.com/joomla/joomla-cms/blob/staging/libraries/cms/application/administrator.php#L6 for example, it is also use 3 space for @copyright tag.

Comment (laoneo) [inline libraries/src/Cms/Controller/Admin.php]: The package information is not required for the namespaced classes. The namespaced classes do have a different doc block than the none namespaced ones.

Comment (joomdonation) [inline libraries/src/Cms/Controller/Admin.php]: OK. I changed the doc block of these classes to make it the same with other namespaced classed. CS looks OK now

File: libraries/src/Cms/Dispatcher/Dispatcher.php [hunk 2]
```diff
@@ -163,11 +163,16 @@ protected function getApplication()
 	 */
 	public function getController($name, $client = null, $config = array())
 	{
-		$client = $client ? $client : ucfirst($this->app->getName()) . '\\';
+		$client = $client ? $client : ucfirst($this->app->getName());
 
-		$controllerName = $this->namespace . $client . 'Controller\\' . ucfirst($name);
+		$controllerClass = $this->namespace . $client . '\\Controller\\' . ucfirst($name);
```
Comment (RonakParmar) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: In old code, '\\' was appended to only ucfirst($this->app->getName()) value, and not to the $client value.
But the current code is appending '\\' to both the values "ucfirst($this->app->getName())" and "$client".
Is it correct to append '\\' to the $client value?

Comment (joomdonation) [inline libraries/src/Cms/Dispatcher/Dispatcher.php]: Yes. That's correct. $client if passed, will have Site or Administrator value, so append '\' is correct

**Comment by RonakParmar** [comment]
I have tested this item :white_check_mark: successfully on b47213c2cfc1da8ce841ca693210cb61c71d55b9<br /><br />Done code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15075">issues.joomla.org/tracker/joomla-cms/15075</a>.</sub>

**Comment by joomdonation** [comment]
So I made further change beside to just changes default controller name;

1. Change Controller to BaseController. It will allows us to write shorter code in default component controller

use Joomla\Cms\BaseController;
class Controller extends BaseController;

Instead of 

use Joomla\Cms\Controller as CmsController;
class Controller extends CmsController;

2. Passing option and name to controller config array. That save controller not having to detect dynamic these information from class name.

**Comment by joomdonation** [comment]
Again, close because I could not resolve conflicts :(
